gems/aws-sigv4/lib/aws-sigv4/signer.rb

```diff
@@ -654,7 +654,9 @@ def uri_escape_path(string)
 
       def fetch_credentials
         credentials = @credentials_provider.credentials
-        if credentials.access_key_id && credentials.secret_access_key
+        # ensure credentials are not blank
```

Comment:
Why not `credentials.set?` If you can't use `.set?` I think regex is a bad idea.. would prefer `nil?` and `empty?`

Reply:
yeah - I was trying to cover the case of blank (eg: '  '), but I agree that's not necessary.  Using `credentials#set?` method now which should work for all credentials/credential providers.